Sources/DatadogSDKTesting/DDTracer.swift

```diff
@@ -54,6 +54,7 @@ internal class DDTracer {
         tracerProvider.updateActiveSampler(Samplers.alwaysOn)
         let spanLimits = tracerProvider.getActiveSpanLimits().settingAttributeCountLimit(1024)
         tracerProvider.updateActiveSpanLimits(spanLimits)
+        tracerProvider.updateActiveClock(NTPClock())
```

Comment:
I don't have enough visibility on how `otel-swift` uses the `Clock.now`, but one thing we had to mitigate in `dd-sdk-ios` was `Kronos.Clock.now` being not monotonic until `.sync()` resolves all calls to the NTP pool.

In other words, this:
```swift
let startTime = Kronos.Clock.now
Thread.sleep(0.1)
let delta =  Kronos.Clock.now - startTime
```
might give negative `delta` for the first attempts when NTP sync is being performed.

When browsing [this](https://github.com/open-telemetry/opentelemetry-swift/blob/eec1f8f1eeec5cf96c3ff76378ec290156707326/Sources/OpenTelemetrySdk/Trace/SpanBuilderSdk.swift#L36) (start) and [this](https://github.com/open-telemetry/opentelemetry-swift/blob/eec1f8f1eeec5cf96c3ff76378ec290156707326/Sources/OpenTelemetrySdk/Trace/RecordEventsReadableSpan.swift#L257) (end) code in `otel-swift`, I'm not sure if this is not the case here. Is it?

Reply:
It's definitely the easiest solution 👍, will work if you can make this threading assumption. Otherwise, we solved this (in https://github.com/DataDog/dd-sdk-ios/pull/607) by [using](https://github.com/DataDog/dd-sdk-ios/blob/84b6512fefea61b158113c83824ed470370ac6ff/Sources/Datadog/Core/System/Time/ServerDateProvider.swift#L27) our [`ValuePublisher`](https://github.com/DataDog/dd-sdk-ios/blob/master/Sources/Datadog/Core/Utils/ValuePublisher.swift) which uses `DispatchQueue`. This comes with lock penalty, which we mitigate by using concurrent queue with `.barrier` for making (rare) writes exclusive and allowing (frequent) simultaneous reads.

Reply:
If I find my solution performance is not good enough I will try to change to your solution, its way better than mine.